### ISSUE TITLE
support 'markevery' in prop_cycle

### ISSUE THREAD

**Issue by hedongya** [matplotlib/matplotlib#8576]
### Bug report

**Bug summary**

Key axes.prop_cycle: Unknown artist properties: {'markevery'}

**Code for reproduction**

```python
# Paste your code here
import matplotlib
from cycler import cycler
matplotlib.rcParamsDefault['axes.prop_cycle']=cycler(markevery=[8])
```

**Actual outcome**

```
ValueError: Key axes.prop_cycle: Unknown artist properties: {'markevery'}
```

**Expected outcome**

Default markevery cycle.

**Matplotlib version**

  * Operating System: Ubuntu 14.04
  * Matplotlib Version: 2.0.0
  * Python Version: 3.4
  * Jupyter Version (if applicable):
  * Other Libraries: 

No error for matplotlib 1.5

**Comment by BowenFu** [comment]
Same problem.

**Comment by tacaswell** [comment]
The issue is that validate the keys in cycler with a white list (see https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/rcsetup.py#L716 ).  The simplest fix is to just extend that to also include markevery, but I am slightly concerned about un-intended consequences.

Exact work:

 - add `markevery` + validator to rcsetup
 - add test of using markevery in the property cycle
 - add whats-new entry

**Comment by WeatherGod** [comment]
Is markevery an artist property? If so, then it should be made into a valid
argument to the property cycler. If not, then adding support for it is very
tricky.

**Comment by BowenFu** [comment]
@WeatherGod I believe it is also a good idea to directly allow 
`matplotlib.rcParamsDefault['lines.markevery'] = 8`

**Comment by RavingMadScientist** [comment]
@tacaswell Hey Thomas, sorry for being so slow on this, but this new validate_markevery method I'm about to PR should enable the full API of lines.set_markevery for the 'axes.prop_cycle' parameter of rcParams, as opposed to restricting assignments to any of the pre-existing validators. If you want to use the example, just lemme know what I need to do to get it into the documentation. Was a pleasure coding with you last weekend!

Cheers :)

**Comment by tacaswell** [comment]
@RavingMadScientist What is the state of this?

**Comment by timhoffm** [comment]
Implemented by #10713
